modules/preview-react/InformationHighlight/lib/parts/Icon.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import {SystemIcon, SystemIconProps, systemIconStencil} from '@workday/canvas-kit-react/icon';
+import {createSubcomponent, ExtractProps} from '@workday/canvas-kit-react/common';
+import {cssVar} from '@workday/canvas-kit-styling';
+import {
+  infoIcon,
+  exclamationCircleIcon,
+  exclamationTriangleIcon,
+} from '@workday/canvas-system-icons-web';
+import {base} from '@workday/canvas-tokens-web';
+import {useInformationHighlightModel} from '../hooks/modelHook';
+
+export interface IconProps extends Omit<Partial<ExtractProps<typeof SystemIcon, never>>, 'icon'> {
+  icon?: SystemIconProps['icon'];
+}
+
+const iconStyles = {
+  informational: {
+    low: systemIconStencil({
+      accentColor: cssVar(base.blueberry400),
+      color: cssVar(base.blueberry400),
+      backgroundColor: 'none',
+    }),
+    high: systemIconStencil({
+      accentColor: cssVar(base.frenchVanilla100),
+      color: cssVar(base.blueberry400),
+      backgroundColor: cssVar(base.blueberry400),
+    }),
+  },
+  caution: {
+    low: systemIconStencil({
+      accentColor: cssVar(base.blackPepper400),
+      color: cssVar(base.blackPepper400),
+      backgroundColor: 'none',
+    }),
+    high: systemIconStencil({
+      accentColor: cssVar(base.frenchVanilla100),
+      color: cssVar(base.blackPepper400),
+      backgroundColor: cssVar(base.blackPepper400),
+    }),
+  },
+  critical: {
+    low: systemIconStencil({
+      accentColor: cssVar(base.cinnamon500),
+      color: cssVar(base.cinnamon500),
+      backgroundColor: 'none',
+    }),
+    high: systemIconStencil({
+      accentColor: cssVar(base.frenchVanilla100),
+      color: cssVar(base.cinnamon500),
+      backgroundColor: cssVar(base.cinnamon500),
+    }),
+  },
+};
+
+const defaultIcons = {
+  informational: infoIcon,
+  caution: exclamationTriangleIcon,
+  critical: exclamationCircleIcon,
+};
+
+export const Icon = createSubcomponent('span')({
+  displayName: 'Icon',
+  modelHook: useInformationHighlightModel,
+})(({icon, ...props}: IconProps, Element, model) => {
+  return (
+    <SystemIcon
+      as={Element}
+      icon={icon ? icon : defaultIcons[model.state.variant]}
+      {...iconStyles[model.state.variant][model.state.emphasis]}
+      {...props}
```

Comment:
```suggestion
      {...handleCsProp(props, informationIconStencil({[model.state.variant]: model.state.emphasis}))}
```

- Use handleCsProp to ensure correct style merging.
- Now you can pass the info to your stencil that it needs, the `variant` and the `emphasis`

Reply:
You'll need to import handleCsProp and createStencil